pycbc/fft/fftw.py

```diff
@@ -381,11 +381,11 @@ def ifft(invec, outvec, prec, itype, otype):
 # translate input and output dtypes into the correct planning function.
 
 _plan_funcs_dict = { ('complex64', 'complex64') : plan_many_c2c_f,
-                     ('complex64', 'float32') : plan_many_r2c_f,
-                     ('float32', 'complex64') : plan_many_c2r_f,
+                     ('float32', 'complex64') : plan_many_r2c_f,
+                     ('complex64', 'float32') : plan_many_c2r_f,
                      ('complex128', 'complex128') : plan_many_c2c_d,
-                     ('complex128', 'float64') : plan_many_r2c_d,
-                     ('float64', 'complex128') : plan_many_c2r_d }
+                     ('float64', 'complex128') : plan_many_r2c_d,
+                     ('complex128', 'float64') : plan_many_c2r_d }
```

Comment:
So this was certainly wrong before, not sure why I ever thought it was right.

As far as "why didn't we see it?", my guess is that we don't actually use the class inferface when performing R2C transforms, or C2R, since that's not in our inner loop.  In particular it does look like the C2C that is the back-bone of our matched filtering was correct, and that is almost certainly why this did not crop up.